examples/github-deploy-only.yml

```
@@ -0,0 +1,82 @@
+name: Build and upload to PyPI
+
+on: [push, pull_request, release]
+
+jobs:
+  build_wheels:
+    name: Build wheels on ${{ matrix.os }}
+    runs-on: ${{ matrix.os }}
+    strategy:
+      matrix:
+        os: [ubuntu-18.04, windows-latest, macos-latest]
+
+    steps:
+    - uses: actions/checkout@v1
+
+    - uses: actions/setup-python@v1
+      name: Install Python
+      with:
+        python-version: '3.7'
+
+    - name: Install cibuildwheel
+      run: |
+        python -m pip install cibuildwheel==1.4.2
+
+    - name: Install Visual C++ for Python 2.7
+      if: startsWith(matrix.os, 'windows')
+      run: |
+        choco install vcpython27 -f -y
+
+    - name: Build wheels
+      run: |
+        python -m cibuildwheel --output-dir wheelhouse
+
+    - uses: actions/upload-artifact@v1
+      with:
+        name: wheels
+        path: ./wheelhouse
+
+  build_sdist:
+    name: Build source distribution
+    runs-on: ubuntu-latest
+    steps:
+    - uses: actions/checkout@v1
+
+    - uses: actions/setup-python@v1
+      name: Install Python
+      with:
+        python-version: '3.7'
+
+    - name: Build sdist
+      run: python setup.py sdist -d wheelhouse/
+
+    - uses: actions/upload-artifact@v2
+      with:
+        path: wheelhouse/*.tar.gz
+
+  upload_pypi:
+    needs: [build_wheels, build_sdist]
+    runs-on: ubuntu-latest
+
+    steps:
+    - uses: actions/setup-python@v2
+      with:
+        python-version: "3.8"
+
+    - name: Install dependencies
+      run: python -m pip install twine
+
+    - uses: actions/download-artifact@v2
+      with:
+        name: artifact
+        path: wheelhouse
```

```suggestion
        path: dist
```

Why not just put the wheels and sdist where twine expects them?

---

I don't really know. This is where they were put in the minimal example as well, and in the examples from the start. Up to @joerick to explain if/why he prefers `wheelhouse` to `dist`?

---

I'm not recommending to move the wheels from wheelhouse, just to make the sdist in dist, and then to download all the wheels and the dist into dist before publish.

---

I think because they are wheelhouses? A wheelhouse contains *all* built wheels. If it had to build a wheel to make the current wheel, they get added here too. So ideally, you may want to do `wheelhouse/<package_name>*.whl` when you upload, do just get the part you want to distribute.

---

A good way to check: require Pandas on Python 2.7. It's missing one of the types of manylinux wheels for 2.7 (`mu`?), so it will probably get built and added to the wheelhouse.

---

And, no, Pandas has no interest in adding missing wheels or touching 2.7 at all. I've asked. :)

---

OK, I misunderstood, then. Thanks! Changed!

> And, no, Pandas has no interest in adding missing wheels or touching 2.7 at all. I've asked. :)

I can't blame them ;-)

---

> I can't blame them ;-)

I've used it to push (somewhat successfully) for some Pandas-dependent projects to be Python 3 only, so it's not only negative ;)